src/Razor/Microsoft.AspNetCore.Razor.Language/src/DefaultBoundAttributeDescriptorBuilder.cs

```diff
@@ -51,6 +51,8 @@ public DefaultBoundAttributeDescriptorBuilder(DefaultTagHelperDescriptorBuilder
 
         public override bool IsDictionary { get; set; }
 
+        public override string DefaultValue { get; set; }
```

Comment:
I'm a bit concerned about us storing the actual default value of TagHelpers. Reason being we serialize this data 😢.

Also, in the `null` case how do you differentiate between these when rendering the default value?
![image](https://user-images.githubusercontent.com/2008729/119576660-0a317e00-bd6e-11eb-8f03-724e547ba5e3.png)

Reply:
> Also, in the null case how do you differentiate between these when rendering the default value?

One is the string literal null and the other is the object null. This scenario is handled in the code below.

https://github.com/dotnet/aspnetcore/blob/9792f12b6f91e87df32658cdb5c0b4ca0789d646/src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ViewComponentTagHelperDescriptorFactory.cs#L208-L220

In both cases, `HasExplicitDefaultValue` is true. For the null object case, ExplicitDefaultValue will be null and NullableAnnotation will be false so we won't right out a default value. For the string literal case, we'll set the default to the string `"null"`.

Reply:
But how would you know how to render the string version of `null`?

Aka how would you generate:
```C#
public string foo { get; set; } = "null";
```

and not 
```C#
public string foo { get; set; } = null;
```